gce_workload_cert_refresh/main.go

```diff
@@ -312,11 +339,16 @@ func refreshCreds() error {
 		return fmt.Errorf("Error rotating target link: %v", err)
 	}
 
-	if oldTarget != "" {
-		logger.Infof("Remove old content dir %s", oldTarget)
-		if err := os.RemoveAll(oldTarget); err != nil {
-			return fmt.Errorf("Failed to remove old symlink target: %v", err)
+	newTarget, err := os.Readlink(symlink)
+	if err == nil {
+		if oldTarget != "" && oldTarget != newTarget {
+			logger.Infof("Remove old content dir %s", oldTarget)
```

Comment:
several different ways we could approach this to make it readable, but i think the primary issue is this kind of code is a product of the strategy used above. suggest reviewing feedback above, revisiting strategy, then this section may become more obvious.

Reply:
this was logic originally to handle the first creation case, which you've changed in this PR. oldtarget can now never be "", since you unconditionally create the symlink on line 276. what are you trying to do here, maybe i can suggest an amendment?